### PR TITLE
Add cursor style to Text

### PR DESCRIPTION
**Pull request by shwanton**
#### Please select one of the following
- [ ] I am removing an existing difference between facebook/react-native and microsoft/react-native-macos :thumbsup:
- [ ] I am cherry-picking a change from Facebook's react-native into microsoft/react-native-macos :thumbsup:
- [x] I am making a fix / change for the macOS implementation of react-native
- [ ] I am making a change required for Microsoft usage of react-native

## Summary

#760 Added support for changing cursor styles for views.
#1346 changed RCTTextView to use NSTextView api's with support for customizing cursor.

Differential Revision: https://phabricator.intern.facebook.com/D28162138

## Change-log

[macOS][Added] - Add cursor styles to Text View

## Test Plan

https://user-images.githubusercontent.com/96719/199860881-4d9138d2-d934-4ea1-bd78-53d3166fd1ff.mp4

### CODE REVIEW

File: Libraries/Text/Text/RCTTextView.m [hunk 1]
```diff
@@ -98,10 +98,6 @@ - (instancetype)initWithFrame:(CGRect)frame
   return self;
 }
 
-#if TARGET_OS_OSX // [TODO(macOS GH#774)
-
-#endif // ]TODO(macOS GH#774)
-
```
[the remark this answers is not in the document]
Comment (shwanton) [reply Libraries/Text/Text/RCTTextView.m:104]: Ah, looks like I removed code in #1346, but forgot to remove the def check! https://github.com/microsoft/react-native-macos/commit/cf48f8d6bc308f8d22ccd6351ef55e7fc4d6eddf

https://github.com/microsoft/react-native-macos/blame/main/Libraries/Text/Text/RCTTextView.m#L101-L103

File: Libraries/Text/VirtualText/RCTVirtualTextViewManager.m [hunk 2]
```diff
@@ -6,7 +6,6 @@
  */
 
 #import <React/RCTVirtualTextViewManager.h>
-
```
[the remark this answers is not in the document]
Comment (shwanton) [reply Libraries/Text/VirtualText/RCTVirtualTextViewManager.m:9]: Inadvertent, will revert.

**Comment by Saadnajmi** [comment]
Any blocker for this one?

**Comment by shwanton** [comment]
This is ready! Just had a lint fix.

**Comment by Saadnajmi** [comment]
> This is ready! Just had a lint fix.

Can you mark it as not a draft so I can merge it? :)

**Comment by shwanton** [comment]
Oops, my bad! 😅

**Comment by christophpurrer** [comment]
Great to see this merged!

**Pull request merged**